### PR TITLE
[IMP] cf: add data bar rule

### PR DESCRIPTION
**Pull request by pro-odoo**
This commit adds the support for the data bar
rule in the conditional format. The data bar is
a visual representation of the value of a cell
in a range of cells. Note that the data bar rule
can be based on the value of the cell itself or
on the value in another cell.

task: 4173340

## Description:

description of this task, what is implemented and why it is implemented that way.

Task: [TASK_ID](https://www.odoo.com/web#id=TASK_ID&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_t("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/plugins/core/conditional_format.ts [hunk 1]
```diff
@@ -60,6 +61,35 @@ export class ConditionalFormatPlugin
...
+      if (rule.rule.type === "DataBarRule" && rule.rule.rangeValues) {
+        const change = applyChange(rule.rule.rangeValues);
+        switch (change.changeType) {
+          case "REMOVE":
+            this.history.update(
+              "cfRules",
+              sheetId,
+              this.cfRules[sheetId].indexOf(rule),
+              "rule",
+              //@ts-expect-error
```
Comment (pro-odoo) [inline src/plugins/core/conditional_format.ts:73]: `rangeValues` does not exist on the union of the `ConditionalFormatRuleInternal`. Do not know how to fix it

File: src/constants.ts [hunk 2]
```diff
@@ -195,6 +195,7 @@ export const DEFAULT_STYLE = {
   fontSize: 10,
   fillColor: "",
   textColor: "#000000",
+  fillColorPercentage: 100,
```
Comment (VincentSchippefilt) [inline src/constants.ts:198]: nope, this belongs to the CF, not to the cell.

File: src/plugins/ui_core_views/evaluation_conditional_format.ts [hunk 3]
```diff
@@ -249,6 +276,53 @@ export class EvaluationConditionalFormatPlugin extends UIPlugin {
...
+    range: string,
+    rule: DataBarRule,
+    computedDataBars: ComputedDataBars
+  ): void {
+    const rangeValues = this.getters.getRangeFromSheetXC(sheetId, rule.rangeValues || range);
+    const allValues = this.getters
+      .getEvaluatedCellsInZone(sheetId, rangeValues.zone)
+      .filter((cell): cell is NumberCell => cell.type === CellValueType.number)
+      .map((cell) => cell.value);
+    const max = Math.max(...allValues);
```
Comment (VincentSchippefilt) [inline src/plugins/ui_core_views/evaluation_conditional_format.ts:290]: use helpers.misc.largeMax instead

File: src/components/side_panel/conditional_formatting/cf_editor/cf_editor.ts [hunk 4]
```diff
@@ -311,6 +319,10 @@ export class ConditionalFormattingEditor extends Component<Props, SpreadsheetChi
           operator: "gt",
         },
       },
+      dataBar: {
+        type: "DataBarRule",
+        color: 0xffffff,
```
Comment (VincentSchippefilt) [inline src/components/side_panel/conditional_formatting/cf_editor/cf_editor.ts:324]: if the default is white, the user is forced to always choose another color wich is an extra step. Can we have a nice light green as default color ?

**Comment by VincentSchippefilt** [comment]
robodoo r+

**Comment by VincentSchippefilt** [comment]
robodoo r-

**Comment by VincentSchippefilt** [comment]
robodoo r+

**Pull request closed without merging**